mpf/devices/multiball_lock.py

```diff
@@ -305,3 +305,26 @@ def _post_events(self, device, **kwargs):
     def _request_new_balls(self, balls):
         """Request new ball to playfield."""
         self.source_playfield.add_ball(balls=balls)
+    
+    def compensate_lock(self):
+        # evilmopey solution for issue 1278
+        if self.config['locked_ball_counting_strategy'] == "no_virtual":
+            self._locked_balls = 0
+            # remove balls that were in lock
+            balance = (self._locked_balls - (-1 * self._physically_locked_balls))  # -1 to make the number positive
+            for i in range(1, balance):
+                self.lock_devices[0].eject()
+        if self.config['locked_ball_counting_strategy'] == "physical_only":
+            self._locked_balls = self.locked_balls(self)
+        if self.config['locked_ball_counting_strategy'] == "virtual_only" or "min_virtual_physical":
+            self._locked_balls = self.locked_balls(self)
+        if self._locked_balls > self._physically_locked_balls:
+             # This means we need to add more balls to the physical lock
+            balance = self._locked_balls - self._physically_locked_balls
+            self._request_new_balls(balance)
```

Comment:
The condition looks right. However, request_new_balls is not the correct function because it will request the ball to the playfield. Instead you would call request_ball on self.lock_devices[x].

Reply:
So I took out the if statements for both no_virtual and physical_only leaving only the if statement that pertains to virtual_only or min_virtual_physical. This if statement only adds balls not takes them away. My only problem now is that on line 9 MYPY is false, this blocks the class BALLDevice from being imported and being able to use the request_ball function. I added the import above it so I could use the request_ball function. 

On a separate note is their a way I can edit this pull request to show you the changes that I described above? Or do I have to send another one with my edited code? Sorry this is my first pull request.